### PR TITLE
[MMMDATA-875] Add GeoPoint target type

### PR DESCRIPTION
**Pull request by nek-taras**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
This is proposal PR to Add GeoPoint target type.

### CODE REVIEW

**Review by nickolai-dr — APPROVED** [review]

File: custom_model_runner/datarobot_drum/drum/data_marshalling.py [hunk 1]
```diff
@@ -40,6 +40,8 @@ def marshal_predictions(
         return _classification_marshal_preds(predictions, request_labels, model_labels)
     elif target_type.is_single_column():
         return _single_col_marshal_preds(predictions)
+    elif target_type.is_multicolumn():
+        return _multi_col_parshal_preds(predictions, request_labels)
```
Comment (yakov-g) [inline custom_model_runner/datarobot_drum/drum/data_marshalling.py:44]:
```suggestion
        return _multi_col_marshal_preds(predictions, request_labels)
```

File: custom_model_runner/datarobot_drum/drum/data_marshalling.py [hunk 2]
```diff
@@ -105,6 +107,10 @@ def _single_col_marshal_preds(predictions):
     return pd.DataFrame(predictions, columns=[PRED_COLUMN])
 
 
+def _multi_col_parshal_preds(predictions, labels):
```
Comment (yakov-g) [inline custom_model_runner/datarobot_drum/drum/data_marshalling.py:110]:
```suggestion
def _multi_col_marshal_preds(predictions, labels):
```

**Review by yakov-g — APPROVED** [review]

File: custom_model_runner/datarobot_drum/drum/adapters/model_adapters/python_model_adapter.py [hunk 3]
```diff
@@ -584,15 +584,16 @@ def _predict_new_drum(self, data, **kwargs) -> RawPredictResponse:
...
+            request_labels = get_request_labels(
                 kwargs.get(CLASS_LABELS_ARG_KEYWORD),
                 positive_class_label,
                 negative_class_label,
             )
-            if self._target_type in {TargetType.BINARY, TargetType.MULTICLASS}
-            else None
-        )
+        elif self._target_type == TargetType.GEO_POINT:
+            request_labels = ["latitude", "longitude"]
```
Comment (akshoop) [inline custom_model_runner/datarobot_drum/drum/adapters/model_adapters/python_model_adapter.py:594]: Consider making new enums for geo point request labels, which would be added in:
```
custom_model_runner/datarobot_drum/drum/enum.py
```

File: custom_model_runner/datarobot_drum/drum/enum.py [hunk 4]
```diff
@@ -391,13 +391,17 @@ class TargetType(Enum):
     MULTICLASS = "multiclass"
     TRANSFORM = "transform"
     TEXT_GENERATION = "textgeneration"
+    GEO_POINT = "geopoint"
```
Comment (akshoop) [inline custom_model_runner/datarobot_drum/drum/enum.py:394]: Based on in-line comment for this class `TargetType` and once you make more development progress for geo point type, make sure to update root-level documentation:
```
MODEL-METADATA.md
```

File: tests/fixtures/geo_point_custom.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+"""
+Copyright 2023 DataRobot, Inc. and its affiliates.
```
Comment (akshoop) [inline tests/fixtures/geo_point_custom.py:2]:
```suggestion
Copyright 2024 DataRobot, Inc. and its affiliates.
```

**Review by akshoop — APPROVED** [review]
LGTM Core Modeling with comments

**Pull request merged**